packages/api/cms-api/src/translation/azure-ai-translator.resolver.ts

```diff
@@ -0,0 +1,45 @@
+import createClient, { isUnexpected, TextTranslationClient } from "@azure-rest/ai-translation-text";
+import { Inject } from "@nestjs/common";
+import { Args, Query, Resolver } from "@nestjs/graphql";
+
+import { RequiredPermission } from "../user-permissions/decorators/required-permission.decorator";
+import { AzureAITranslatorConfig } from "./azure-ai-translator.config";
+import { AZURE_AI_TRANSLATOR_CONFIG } from "./azure-ai-translator.constants";
+import { TranslationInput } from "./dto/translation.input";
+
+@Resolver()
+@RequiredPermission(["translation"], { skipScopeCheck: true })
+export class AzureAITranslatorResolver {
+    private translationClient: TextTranslationClient;
+    constructor(@Inject(AZURE_AI_TRANSLATOR_CONFIG) private readonly config: AzureAITranslatorConfig) {
+        this.translationClient = createClient(config.endpoint, {
+            key: config.key,
+            region: config.region,
+        });
+    }
+    @Query(() => String)
+    async translate(@Args("input") input: TranslationInput): Promise<string> {
```

Comment:
This is too generic for a library resolver. Suppose we want to add a DeepL translation service in the future, then the query names would clash. I'd change it to `azureAiTranslate(input: AzureAiTranslationInput`. @thomasdax98 what do you think?

Reply:
Yes, I also think `azureAiTranslate(input: AzureAiTranslationInput)` is better